### PR TITLE
Main menu: make title char *

### PR DESCRIPTION
**Pull request by vlolteanu**

### CODE REVIEW

**Comment by AJenbo** [comment]
What's the rational behind this?

**Comment by vlolteanu** [comment]
The two instances of "title" always contain stuff obtained from LanguageTranslate, or an empty string. It makes no sense to sense to strcpy an immutable string, hence const char * is appropriate.

It's similar to what I've done here: https://github.com/diasurgical/devilutionX/pull/1750

**Review by AJenbo — APPROVED** [review]

File: Source/DiabloUI/selgame.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ extern int provider;
 
 namespace {
 
-char title[32];
+const char *title = "";
```
Comment (julealgon) [inline Source/DiabloUI/selgame.cpp:38]: Can't we use `string` instead?

Comment (vlolteanu) [reply Source/DiabloUI/selgame.cpp:38]: That would just make extra copies. If you really want to use some something from the stl, `string_view` is the closest to what we're doing.

Comment (julealgon) [reply Source/DiabloUI/selgame.cpp:38]: Can we use `string_view` then?

Comment (AJenbo) [reply Source/DiabloUI/selgame.cpp:38]: Eventually all of it should be moved to `string`/`string_view`, it's already what is being used internally in the translation handling as well as the `fmt` string building. Most of the warnings we get in MSVC is from the use of `char*` for string handling. I guess the next step in this process would be to adapt the string rendering code to expect `std::string`, and then take out all the in between conversions as a final stage.

**Comment by jmechnich** [comment]
This PR broke menu titles (see #2373).

**Pull request merged**